File: BaseLib/src/main/java/com/hjc/baselib/event/EventManager.java
package com.hjc.baselib.event;

import org.greenrobot.eventbus.EventBus;

/**
 * @Author: HJC
 * @Date: 2019/1/4 14:57
 * @Description: EventBus管理类
 */
public class EventManager {

    public static void register(Object subscriber) {
        if (!EventBus.getDefault().isRegistered(subscriber)) {
            EventBus.getDefault().register(subscriber);
        }
    }

    public static void unregister(Object subscriber) {
        if (EventBus.getDefault().isRegistered(subscriber)) {
            EventBus.getDefault().unregister(subscriber);
        }
    }

    public static void sendEvent(MessageEvent event) {
        EventBus.getDefault().post(event);
    }

    public static void sendStickyEvent(MessageEvent event) {
        EventBus.getDefault().postSticky(event);
    }

}

File: BaseLib/src/main/java/com/hjc/baselib/event/EventManager.kt
package com.hjc.baselib.event

import org.greenrobot.eventbus.EventBus

/**
 * @Author: HJC
 * @Date: 2019/1/4 14:57
 * @Description: EventBus管理类
 */
object EventManager {

    fun register(subscriber: Any?) {
        if (!EventBus.getDefault().isRegistered(subscriber)) {
            EventBus.getDefault().register(subscriber)
        }
    }

    fun unregister(subscriber: Any?) {
        if (EventBus.getDefault().isRegistered(subscriber)) {
            EventBus.getDefault().unregister(subscriber)
        }
    }

    fun sendEvent(event: MessageEvent<*>?) {
        EventBus.getDefault().post(event)
    }

    fun sendStickyEvent(event: MessageEvent<*>?) {
        EventBus.getDefault().postSticky(event)
    }
}